### PR TITLE
Handle mocking objects for both PHPUnit 4 and 5

### PR DESCRIPTION
**Pull request by piotr-zuralski**
- avoid deprecations on mocks
- require PHPUnit less than 6.0

new PHPUnit 6.0 does not support tap parameter, TAP format is not supported any more in PHPUnit 6 - https://github.com/sebastianbergmann/phpunit/blob/master/ChangeLog-6.0.md#removed

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     },
     "require": {
         "php": ">=5.4.0",
-        "phpunit/phpunit": "^4.5|^5.0",
+        "phpunit/phpunit": "^4.5|^5.0|<=6.0",
```
Comment (piotr-zuralski) [inline composer.json]: https://getcomposer.org/doc/articles/versions.md#caret

File: tests/Adapter/Phpunit/XmlConfigurationTest.php [hunk 2]
```diff
@@ -15,9 +15,12 @@
 use Humbug\Adapter\Locator;
 use Humbug\Adapter\Phpunit\ConfigurationLoader;
 use Humbug\Adapter\Phpunit\XmlConfiguration;
+use Humbug\Test\PHPUnitHandleDeprecatedMethodsTrait;
 
 class XmlConfigurationTest extends \PHPUnit_Framework_TestCase
 {
+    use PHPUnitHandleDeprecatedMethodsTrait;
```
Comment (GrahamCampbell) [inline tests/Adapter/Phpunit/XmlConfigurationTest.php]: Kill phpunit 4.

Comment (piotr-zuralski) [inline tests/Adapter/Phpunit/XmlConfigurationTest.php]: ok :+1:

**Comment by piotr-zuralski** [comment]
@GrahamCampbell with dropping support for PHPUnit 4 we will drop support for PHP 5.4 and 5.5

**Comment by GrahamCampbell** [comment]
> @GrahamCampbell with dropping support for PHPUnit 4 we will drop support for PHP 5.4 and 5.5

I'm fine with that. I think PHP 5.6 is a good minimum version to use for the next 2 years or so.

**Pull request merged**